src/Account/components/AccountTransactions.tsx

```diff
@@ -87,6 +95,23 @@ function AccountTransactions(props: { account: Account }) {
     router
   ])
 
+  const decodeTx = (envelopeXdr: string): Transaction => {
+    const decoded = TransactionBuilder.fromXDR(envelopeXdr, account.testnet ? Networks.TESTNET : Networks.PUBLIC)
+    return decoded instanceof FeeBumpTransaction ? decoded.innerTransaction : decoded
+  }
+
+  // TODO: keep decoded transaction in app cache, now only `recentTxs.transactions` are cached
+  const txs = React.useMemo(
+    () =>
+      recentTxs.transactions.map(tx => ({
+        ...tx,
+        decodedTx: decodeTx(tx.envelope_xdr)
+      })),
+    [recentTxs.transactions]
+  )
+
+  const filteredTxs = React.useMemo(() => txs.filter(excludeClaimableFilter), [txs])
```

Comment:
```suggestion
  const filteredTxs = React.useMemo(() => txs.filter(excludeClaimableFilter), [txs])

  React.useEffect(() => {
    if (!filteredTxs?.length && recentTxs.olderTransactionsAvailable) {
      handleFetchMoreTransactions()
    }
  }, [filteredTxs, recentTxs.olderTransactionsAvailable, recentTxs.transactions.length])
```
Load more transactions if list is empty after filtration

Reply:
I've done the similar thing differently (in a separate hook). Please see PR updates